parse/quote_test.go

```diff
@@ -24,6 +24,7 @@ func TestUnquote(t *testing.T) {
 		{`'a'`, "a"},
 		{`'\n'`, "\n"},
 		{`'\u2222'`, "\u2222"},
+		{`'\\'`, "\\"},
```

Comment:
wondering if we need another test based on your change to test escaping _after_ a block of test, e.g.
```
{`'\\\n'`, "\\\n"},
```